sycl/test/matrix/matrix-bfloat16-test-coord-gemm.cpp

```diff
@@ -0,0 +1,223 @@
+             auto [row, col] =
+                 sycl::ext::intel::experimental::matrix::get_coord(dataItem);
+             // get_coord_ref(i, spmd_item.get_local_id(1));
+             global_index = row + global_idx * TM;
+
+             sum_local_rows[global_index] += data[i];
+
+             handled_rows[global_index] = 1;
+           }
+
+           for (int j = 0; j < M; j++) {
+             if (handled_rows[j] == 1) {
+               global_index = j;
+               sum_local_rows[global_index] = reduce_over_group(
+                   sg, sum_local_rows[global_index], sycl::plus<>());
+               // only Groups leader perform the global reduction
+               if (global_idy % SG_SZ == 0) {
+                 sycl::atomic_ref<float, sycl::memory_order::relaxed,
+                                  sycl::memory_scope::device>
+                     aref(v[global_index]);
+                 aref.fetch_add(sum_local_rows[global_index]);
+               }
+             }
+           }
+         }); // parallel for
+   }).wait();
+}
+
+bfloat16 A[MATRIX_M][MATRIX_K];
+bfloat16 B[MATRIX_K / 2][MATRIX_N * 2];
+float C[MATRIX_M][MATRIX_N];
+float D[MATRIX_M][MATRIX_N];
+
+float make_fp32(bfloat16 x) {
+  unsigned int y = *((int *)&x);
+  y = y << 16;
+  float *res = reinterpret_cast<float *>(&y);
+  return *res;
+}
+
+void matrix_multiply_ref(int *A_mem, int *B_mem, int *C_mem, int M, int N,
+                         int K) {
+  for (int m = 0; m < M; m++)
+    for (int n = 0; n < N; n++) {
+      for (int k = 0; k < K; k++) {
+        // Because B was assumed VNNIed
+        bfloat16 *va = (bfloat16 *)(A_mem + m * K + k);
+        bfloat16 *vb = (bfloat16 *)(B_mem + k * N + n);
+        float acc = *((float *)(C_mem + m * N + n));
+        for (int i = 0; i < 2; i++) {
+          acc += (make_fp32(va[i]) * make_fp32(vb[i]));
+        }
+        *((float *)(C_mem + m * N + n)) = acc;
+      }
+    }
+}
+
+int main() {
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_K; j++) {
+      A[i][j] = bfloat16(1.0f * (i + j));
+    }
+  }
+  for (int i = 0; i < MATRIX_K / 2; i++) {
+    for (int j = 0; j < MATRIX_N * 2; j++) {
+      B[i][j] = bfloat16(2.0f * i + 3.0f * j);
+    }
+  }
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      C[i][j] = 1.0;
+      D[i][j] = 1.0;
+    }
+  }
+
+  big_matrix<float, MATRIX_M, MATRIX_N> MC((float *)&C);
+  big_matrix<float, MATRIX_M, MATRIX_N> MD((float *)&D);
+  big_matrix<bfloat16, MATRIX_M, MATRIX_K> MA((bfloat16 *)&A);
+  big_matrix<bfloat16, MATRIX_K / 2, MATRIX_N * 2> MB((bfloat16 *)&B);
+  matrix_multiply(MC, MA, MB);
+  matrix_multiply_ref((int32_t *)A, (int32_t *)B, (int32_t *)D, MATRIX_M,
+                      MATRIX_N, MATRIX_K / 2);
+
+  bool res = true;
+  float sum_rows_ref[MATRIX_M] = {0};
+
+  for (int i = 0; i < MATRIX_M; i++) {
+    for (int j = 0; j < MATRIX_N; j++) {
+      // std::cout << C[i][j] << " ";
```

Comment:
remove comment

Reply:
File removed, as we have this in test-suite